sips/sip-030/sip-030-wallet-interface.md

```diff
@@ -0,0 +1,532 @@
+
+Parameter properties
+
+- `recipient`: `string` address, Stacks c32-encoded
+- `asset`: `string` address, Stacks c32-encoded, with contract name suffix
+- `assetId`: `ClarityValue`
+
+`where`
+
+- `ClarityValue`: `string | object` hex-encoded or JSON representation
+
+Result properties
+
+- `txid`: `string` hex-encoded
+- `transaction`: `string` hex-encoded raw transaction
+
+#### Method `stx_callContract`
+
+Parameter properties
+
+- `contract`: `string.string` address with contract name suffix, Stacks c32-encoded
+- `functionName`: `string`
+- `functionArgs`: `ClarityValue[]`, defaults to `[]`
+  - where `ClarityValue` is `string | object` hex-encoded or JSON representation
+
+Result properties
+
+- `txid`: `string` hex-encoded
+- `transaction`: `string` hex-encoded raw transaction
+
+#### Method `stx_deployContract`
+
+Parameter properties
+
+- `name`: `string`
+- `clarityCode`: `string` Clarity contract code
+- `clarityVersion?`: `number`
+
+Result properties
+
+- `txid`: `string` hex-encoded
+- `transaction`: `string` hex-encoded raw transaction
+
+#### Method `stx_signTransaction`
+
+Parameter properties
+
+- `transaction`: `string` hex-encoded raw transaction
+
+Result properties
+
+- `transaction`: `string` hex-encoded raw transaction (signed)
+
+#### Method `stx_signMessage`
+
+Parameter properties
+
+- `message`: `string`
+
+Result properties
+
+- `signature`: `string` hex-encoded
+- `publicKey`: `string` hex-encoded
+
+#### Method `stx_signStructuredMessage`
+
+Parameter properties
+
+- `message`: `string` Clarity value, hex-encoded
+- `domain`: `string` hex-encoded (defined by SIP-018)
+
+Result properties
+
+- `signature`: `string` hex-encoded
+- `publicKey`: `string` hex-encoded
+
+#### Method `stx_getAddresses`
+
+Result properties
+
+- `addresses`: `{}[]`
+  - `address`: `string` address, Stacks c32-encoded
+  - `publicKey`: `string` hex-encoded
+
+#### Method `stx_getAccounts`
+
+> **Comment**: This method is similar to `stx_getAddresses`.
+> It was added to provide better backwards compatibility for applications using Gaia.
+
+Result properties
+
+- `accounts`: `{}[]`
+  - `address`: `string` address, Stacks c32-encoded
+  - `publicKey`: `string` hex-encoded
+  - `gaiaHubUrl`: `string` URL
+  - `gaiaAppKey`: `string` hex-encoded
+
+#### Method `stx_updateProfile`
+
+Parameter properties
+
+- `profile`: `object` Schema.org Person object[^13]
+
+Result properties
+
+- `profile`: `object` updated Schema.org Person object[^13]
+
+### Listeners
+
+In addition to the request interface, event listeners may be provided via the `.listen` method.
+
+- `provider.listen(event: string, listener: (...args: any[]) => void): Function`
+
+> `provider.listen` should return a "unlisten" function, which can be called to remove the listener.
+
+The event name should be closer to nouns than verbs and doesn't use the `on` prefix from DOM naming conventions.
+
+#### Event `accountChange`
+
+`listener: (accounts: {}[]) => void`
```

Comment:
Wondering if this is a privacy issue. A user might only wish to share one particular account details with a given app, but this switcher api implicitly shares all account details, (if a user switches to all accounts).

Instead, perhaps we should return only the account index, such that the app can know the account has changed.